code/modules/mob/dead/observer/observer.dm

```diff
@@ -176,17 +176,6 @@ Works together with spawning an observer, noted above.
 		ghost.can_reenter_corpse = can_reenter_corpse
 		ghost.timeofdeath = src.stat == DEAD ? src.timeofdeath : world.time
 
-
-		//This is duplicated for robustness in cases where death might not be called.
-		//It is also set in the mob/death proc
-		if (isanimal(src))
-			set_death_time(ANIMAL, world.time)
-		else if (ispAI(src) || isDrone(src))
-			set_death_time(MINISYNTH, world.time)
-		else
-			set_death_time(CREW, world.time)//Crew is the fallback
-
-

```

Comment:
I admit I am not entirely sure about deleting this myself, given the comment. But it does seem to properly fix the respawn time resetting. Respawn times on observing also seem to work fine with this gone.

Reply:
If you want to do this, you would need to make sure set_death_time is called everywhere where ghostize is called and it makes sense to call set_death time.

I quickly checked a few procs and I think I found a instance where set_death_time is not called, when it should.
/mob/living/silicon/ai/proc/do_wipe_core()

Reply:
Aha. What about a check over the deleted block making sure it will not be called twice? I imagine that could be more problem-proof.